File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function(root, val) {
  const originalRoot = root;
  if (!root) return new TreeNode(val);

  while (root) {
    if (root.val > val) {
      if (!root.left) {
        root.left = new TreeNode(val);

        return originalRoot;
      }
      
      root = root.left;
    } else {
      if (!root.right) {
        root.right = new TreeNode(val);

        return originalRoot;
      }
      
      root = root.right;
    }
  }
}

File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
      if not root:
        return TreeNode(val)

      original_root = root

      while root:
        if root.val > val:
          if not root.left:
            root.left = TreeNode(val)
          
            return original_root
          
          root = root.left
        else:
          if not root.right:
            root.right = TreeNode(val)
            
            return original_root
          
          root = root.right

        